src/commands/new.rs

```diff
@@ -1,75 +1,90 @@
-use std::path::PathBuf;
-
 use clap::{Parser};
 use itertools::Itertools;
-
+use std::path::PathBuf;
 use crate::hub_api;
+use anyhow::{Context, Result};
+use std::io::{self, Write};
+
 
 #[derive(Parser, Debug)]
 #[clap(about = "Create an application from a template on the Hub")]
 pub struct NewCommand {
     #[clap(short = 't')]
     terms: Vec<String>,
 
-    name: String,
+    #[clap(name = "name", help = "Name of the application to create from the template")]
+    name: Option<String>,
 }
 
 impl NewCommand {
-    pub async fn run(&self) -> anyhow::Result<()> {
+    pub async fn run(&self) -> Result<()> {
+        
         let Some(index_entry) = self.resolve_selection().await? else {
             return Ok(());
         };
-
+    
         println!("Template {} by {}", index_entry.title(), index_entry.author());
         println!("{}", index_entry.summary());
-
+    
+        let app_name = if let Some(ref name) = self.name {
+            name.clone()
+        } else {
+            print!("Enter a name for your new application: ");
```

Comment:
Use `dialoguer::Input` for prompting.  You can see examples in the Spin `templates` crate.